packages/terra-disclosure-manager/src/DisclosureManager.jsx

```diff
@@ -0,0 +1,351 @@
+    const disclosureKeys = Object.assign([], this.state.disclosureComponentKeys);
+
+    /**
+     * Before closing the disclosure, the dismiss checks for components in the stack are
+     * executed in stack order. Components will be dismissed in order up until a rejection occurs, at which point
+     * the blocking component will be presented.
+     */
+    return this.resolveDismissChecksInSequence(disclosureKeys).then(() => {
+      this.dismissChecks = {};
+      this.closeDisclosure();
+    });
+  }
+
+  generatePopFunction(key) {
+    return () => {
+      let promiseRoot = Promise.resolve();
+
+      const dismissCheck = this.dismissChecks[key];
+      if (dismissCheck) {
+        promiseRoot = dismissCheck();
+      }
+
+      return promiseRoot
+        .then(() => {
+          this.dismissChecks[key] = undefined;
+          this.resolveDismissPromise(key);
+        })
+        .then(() => {
+          this.popDisclosure();
+        });
+    };
+  }
+
+  renderContentComponents() {
+    const { children, app, supportedDisclosureTypes } = this.props;
+
+    const childApp = AppDelegate.clone(app, {
+      disclose: (data) => {
+        if (supportedDisclosureTypes.indexOf(data.preferredType) >= 0 || !app) {
+          return this.safelyCloseDisclosure()
+            .then(() => {
+              this.openDisclosure(data);
+              /**
+               * The disclose Promise chain is resolved with a set of APIs that the disclosing content can use to
+               * manipulate the disclosure, if necessary.
+               */
+              return {
+                /**
+                 * The afterDismiss value is a deferred Promise that will be resolved when the disclosed component is dismissed.
+                 */
+                afterDismiss: new Promise((resolve) => {
+                  this.onDismissResolvers[data.content.key] = resolve;
+                }),
+                /**
+                 * The dismissDisclosure value is a function that the disclosing component can use to manually close the disclosure.
+                 * Any and all dismiss checks are still performed.
+                 */
+                dismissDisclosure: this.safelyCloseDisclosure,
+              };
+            });
+        }
+        return app.disclose(data);
+      },
+    });
+
+    return React.Children.map(children, child => React.cloneElement(child, {
+      app: childApp,
+    }));
+  }
+
+  renderDisclosureComponents() {
+    const { app, supportedDisclosureTypes } = this.props;
+    const { disclosureComponentKeys, disclosureComponentData, disclosureIsMaximized, disclosureSize } = this.state;
+
+    return disclosureComponentKeys.map((componentKey, index) => {
+      const componentData = disclosureComponentData[componentKey];
+      const isFullscreen = disclosureSize === 'fullscreen';
+
+      const popContent = this.generatePopFunction(componentData.key);
+      const appDelegate = AppDelegate.create({
+        disclose: (data) => {
```

Comment:
Maybe break off creating this function into a helper function to reduce the size of this method?